python/parquet_tools/csv2pq/csv2csv.py

```diff
@@ -0,0 +1,181 @@
+
+    # *************************************************************************
+    # *                                r e a d                                *
+    # *************************************************************************
+
+    def read(self, size):
+        """Mimic the standard file read() function.
+
+        Parameters
+        ----------
+        size : 'int'
+            The number of bytes to read and return (ignored, see notes).
+
+        Returns
+        -------
+        result : 'string'
+            The manipulated next row of the file is returned unless no more
+            records remain, in which case, a null string is returned.
+
+        Notes
+        -----
+        - Since only we read a single row at a time from the csv file, the
+          caller (a pandas csv reader) knows how to handle varying length
+          records regardless of how much was requested. So, we ignore the
+          size argument.
+        """
+
+        # Get next row unless we have no more rows, then we are done.
+        #
+        row = []
+        try:
+            row = next(self.csvrdr)
+        except StopIteration:
+            return ''      # On EOF return the null string
+        except Exception as exc:  # We don't care what it is
+            raise FatalError(exc, 'Unable to convert csv row', self.nRow,
+                  'in file', self.csvF.name)
+
+        # Validate we have the correct number of columns.
+        #
+        n = 0
+        for r in row:
+            n += 1
+        if Csv2Csv._nCol > 0:
+            if len(row) != Csv2Csv._nCol:
+                raise ParmError(7, 'In file "' + self.csvF.name + '" record',
+                      self.nRow, 'has', len(row),
+                      'columns but schema defines only',
+                      Csv2Csv._nCol, 'columns')
+
+        # Look at all columns that we need to preprocess null values
+        #
+
+        for i in Csv2Csv._nilC:
+            if row[i] == Csv2Csv._nilV:
+                row.append('true')
+                row[i] = Csv2Csv._nanV
+            else:
+                row.append('false')
+        rec = Csv2Csv._sepC.join(row)+'\n'
+
+        # We return one row at a time. We could return more but memory
+        # reallocation is more costly then a call for another record.
+        #
+        return rec
+
+    # *************************************************************************
+    # *                               s e t u p                               *
+    # *************************************************************************
+    @classmethod
+    def setup(cls, cmdinfo, schema):
+        """Setup this class for future use.
+
+        Parameters
+        ----------
+        cmdinfo : 'object'
+            The command information object holding various options.
+        schema : 'object'
+            The schema object defining the csv file schema.
+
+        Returns
+        -------
+        result : 'None'
```

Comment:
formatting.
but I think you don't need `Returns` if a function doesn't return anything?

Reply:
What formatting are you talking about here?

Reply:
Ah, the email had the rest of the sentence which oddly did not appear here. The missing text was: "but I think you don't need `Returns` if a function doesn't return anything?" and my response is that I've seen it done both ways. There seems to be large amount of silence on what you should do. A conservative person would document it to simply prove that there was no omission.